userspace/falco/outputs_http.cpp

```diff
@@ -34,15 +34,58 @@ void falco::outputs::output_http::output(const message *msg)
 		} else {
 			slist1 = curl_slist_append(slist1, "Content-Type: text/plain");
 		}
+		res = curl_easy_setopt(curl, CURLOPT_HTTPHEADER, slist1);
 
-		curl_easy_setopt(curl, CURLOPT_HTTPHEADER, slist1);
-		curl_easy_setopt(curl, CURLOPT_URL, m_oc.options["url"].c_str());
-		curl_easy_setopt(curl, CURLOPT_POSTFIELDS, msg->msg.c_str());
-		curl_easy_setopt(curl, CURLOPT_USERAGENT, m_oc.options["user_agent"].c_str());
-		curl_easy_setopt(curl, CURLOPT_POSTFIELDSIZE, -1L);
+		if(res == CURLE_OK)
```

Comment:
Thanks for checking all the results. I was wondering if there was a more elegant way to attain this without all this if-chain, but if there is not I'm ok in going this route.

Reply:
This is the best I came out with. 

Most of these options could be set only once at the creation time of the `http_output` object and reuse the same `curl` object when sending output to the `http endpoint`. We instantiate and initialize a new `curl` object each time we emit a message. Maybe @leogr has more info on why we doing things this way. 
Anyway, this refactoring/optimization will be for another PR.